libs/s25main/world/GameWorld.cpp

```diff
@@ -250,20 +250,42 @@ void GameWorld::BuildRoad(const unsigned char playerId, const bool boat_road, co
         RecalcBQForRoad(end);
         end = GetNeighbour(end, i);
 
-        // Evtl Zierobjekte abreißen
+        // Tear off decorative objects if necessary
         if(HasRemovableObjForRoad(end))
             DestroyNO(end);
     }
 
     auto* rs = new RoadSegment(boat_road ? RoadType::Water : RoadType::Normal, GetSpecObj<noFlag>(start),
                                GetSpecObj<noFlag>(end), route);
-
+    
     GetSpecObj<noFlag>(start)->SetRoute(route.front(), rs);
     GetSpecObj<noFlag>(end)->SetRoute(route.back() + 3u, rs);
 
-    // Der Wirtschaft mitteilen, dass eine neue Straße gebaut wurde, damit sie alles Nötige macht
+   
+
+    // Telling the economy that a new road has been built so that it does whatever it takes
     GetPlayer(playerId).NewRoadConnection(rs);
     GetNotifications().publish(RoadNote(RoadNote::Constructed, playerId, start, route));
+
+    // Addon is enabled, road is not waterway and player is human then add flags to new roadsegment
+    if(GetGGS().isEnabled(AddonId::AUTOFLAGS) && rs->GetRoadType() != RoadType::Water && GetPlayer(playerId).isHuman())
+    {     
+
+        MapPoint nextPT = GetSpecObj<noFlag>(start)->GetPos();
+
+        for(unsigned short i = 0; i < route.size(); ++i)
+        {
+            nextPT = GetNeighbour(nextPT, route[i]);
+
+            if(!IsFlagAround(nextPT))
+            {
+                SetFlag(nextPT, playerId);
+            }
+        }
+    }
+    
+
+        
```

Comment:
Better use a range-based for-loop, and better name for the point (in camelcase "Pt") as it is the "current" on the road not the "next" point

Also avoid empty lines where not required or helpful for readability:
```suggestion
        MapPoint roadPt = GetSpecObj<noFlag>(start)->GetPos();
        for(const Direction curDir: route)
        {
            roadPt = GetNeighbour(roadPt, curDir);
            if(!IsFlagAround(roadPt))
                SetFlag(roadPt, playerId);
        }
    }
```